### DISCUSSION TITLE
ssh -i not passing identity files from command line

### DISCUSSION THREAD

**Discussion by chrisbennight** [blinksh/blink#1891] [Support] [answered]
On an ipad pro 12.9 m1

1. Create a ssh key, call it 'dakey' in the app through the config menu
2. Create a hosts entry for device 
3. ssh in with user/pass, add key to authorized_keys
4. type `ssh host` - where host is the host configured by name - works
5. type `ssh -i dakey username@local.ip.add.ress` and i get prompted for a password

dakey is same as key configured for the host
port is default, 22
username is same in cli or config'd host

i have this same issue on a local ubuntu 22.04 box, local synology nas, and cloud runpod server

the identity file (dakey in my example) is not a path, but the name of the key generated in the app.  if i change this to a non-existent key I still get prompted for a password with no error, so I'm not sure what is going on.

docs indicate i should pass an identity_file, but no indication of what that is - i just assume it's the same named key i use when i config a host in the config menu.

**Comment by carloscabanero** [comment]
Thanks for the thorough report. Your command is correct. Could you send a -vvv dump? In particular, want to see if the connection is trying the public key with the remote and if the remote is discarding it.

I'm assuming that configuring the key as part of the Host works, correct?

**Comment by chrisbennight** [comment, marked as answer]
Well, adding the -vvv I think id'd my issue.

I was cutting and pasting from runpod, which puts flags after the host - `ssh root@ip.ad.dre.ss -p 47151 -i dakey` for example
(so my repro was not accurate, mea culpa)

when -vvv did not work at the end either it clicked, and I realized here ordering of flags and arguments seem to matter (just wasn't used to it)

so thanks 😊

**Reply by chrisbennight** [reply]
(and to be clear, ordering flags before user@host resolved my issue)

**Reply by carloscabanero** [reply]
When we did the rewrite to use Swift's args parsing, it allowed to place the params wherever. But we may have changed something in the structure and you are right those don't work anymore. Will take a look.

**Reply by chrisbennight** [reply]
thanks, seems minor but would def appreciate if the param order thing could be fixed;  it's a definite quality of life frustration mostly from online hosts that present dynamic connection info with params after the host.

**Reply by carloscabanero** [reply]
Just fixed it this week! Should be out on 17.2.0, will release on TF this coming Monday, and probably for everyone before end of month.
